apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java

```diff
@@ -318,26 +338,29 @@ public Bundle findByPatient(
               value = OpenAPIContentProvider.EOB_INCLUDE_TAX_NUMBERS_VALUE)
           String taxNumbers,
       RequestDetails requestDetails) {
+
     /*
      * startIndex is an optional parameter here because it must be declared in the
      * event it is passed in. However, it is not being used here because it is also
      * contained within requestDetails and parsed out along with other parameters
      * later.
      */
-
-    Long beneficiaryId = Long.parseLong(patient.getIdPart());
-    Set<ClaimTypeV2> claimTypes = parseTypeParam(type);
+    OffsetLinkBuilder startIx = new OffsetLinkBuilder(requestDetails, "startIndex");
```

Comment:
so here is where our code/documentation was really confusing; here is comment in current codebase:
    /*
     * startIndex is an optional parameter here because it must be declared in the
     * event it is passed in. However, it is not being used here because it is also
     * contained within requestDetails and parsed out along with other parameters
     * later.
     */

So it implies that sure, one can pass in the 'startIndex' as a URL param, but it won't be used! Only the 'startIndex' in the HTTP Request header will be recognized. In fact, the current OffsetLinkBuilder is where it resides AND that class throws an Exception on the 'getStartIndex' if the 'startIndex' is < 0.

We don't need 2 OffsetLinkBuilder instances here...will fix that. But I still think that we should handle the 'startIndex' value check at time of construction (like we do, the 'pageSize').

Reply:
Yup I think I wrote that comment when I was writing some tests and noticed that startIndex just gets ignored as a passed parameter. I also think it's fine to leave it, and here's why: startIndex when used via the actual API gets translated by HAPI FHIR into the requestDetails value. The only time I think we use startIndex from the spring param is in our testing where we're touching the method directly. Since we can test using requestDetail's index, and that is _the path_ that gets used, I think it's fine to just note that the startIndex is basically only on the param list for documentation purposes. (It's also out of scope of this PR and should probably be fixed as a separate tech debt issue if we do want to fix it)